app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -262,6 +269,29 @@ def remove_disk_config_spec(vim_obj, vmcs, hardware, options)
     end
   end
 
+  def resize_disk_config_spec(vim_obj, vmcs, hardware, options)
+    raise "resize_disk_config_spec: disk filename is required." unless options[:disk_name]
+
+    device = vim_obj.getDeviceByBacking(options[:disk_name], hardware)
+    raise "resize_disk_config_spec: no virtual device associated with: #{options[:disk_name]}" unless device
+    raise "resize_disk_config_spec: decrease size is not supported for: #{options[:disk_name]}" unless device.capacityInKB.to_i <= options[:disk_size_in_mb].to_i * 1024
+
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Edit) do |vdcs|
+      vdcs.device = VimHash.new("VirtualDisk") do |dev|
+        dev.key           = device.key
+        dev.capacityInKB  = options[:disk_size_in_mb].to_i * 1024
+        dev.controllerKey = device.controllerKey
+        dev.unitNumber    = device.unitNumber
+
+        dev.backing = VimHash.new("VirtualDiskFlatVer2BackingInfo") do |bck|
```

Comment:
There's no guarantee that the backing type will be FlatVer2, you have the device so you could either do `VimHash.new(device.backing.xsiType)`, or just use the backing `dev.backing = device.backing` since you aren't changing anything in the backing.

I assume you tried not passing the backing in at all and it failed?

Reply:
Yes, not passing the backing results in --> Error: Incompatible device backing specified for device '0'.

Reply:
👍